dbfit-java/informix/src/test/java/dbfit/environment/InformixBigDecimalTransformerTest.java

```diff
@@ -0,0 +1,45 @@
+package dbfit.environment;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import java.math.BigDecimal;
+import java.sql.SQLException;
+
+import dbfit.util.NormalisedBigDecimal;
+
+public class InformixBigDecimalTransformerTest {
+
+    @Test
+    public void transformsTest() {
+        InformixBigDecimalTransformer bdt = new InformixBigDecimalTransformer();
+        BigDecimal inBigDec = new BigDecimal("123.45");
+        NormalisedBigDecimal nbd = new NormalisedBigDecimal(inBigDec);
+        BigDecimal outBigDec = null;
+        try {
+            outBigDec = (BigDecimal) bdt.transform(nbd);
+        } catch (SQLException e) {
+            fail("InformixBigDecimalTransformer threw SQLException transforming NormalisedBigDecimal input");
+        }
+        assertEquals(outBigDec, inBigDec);
+    }
+
+    @Test
+    public void rejectsInputTest() {
+        InformixBigDecimalTransformer bdt = new InformixBigDecimalTransformer();
+        BigDecimal inBigDec = new BigDecimal("123.45");
+        boolean caught = false;
+        String expectedMsg = "InformixBigDecimalTransformer cannot transform objects of type java.math.BigDecimal";
```

Comment:
> I think that the direction element can be omitted from the actual classes implementing TypeTransformer and instead placed in the TypeTransformerFactory object instance name. E.g. dbfitToJDBCTransformerFactory (this clearly states the source and target for the collection).

Well, it's good the name to reflect what the class is doing. And I don't deny - this also applies to the factory instance name. Let's say we apply the `omit direction` principle and we don't call the in-going transformations `*Normaliser` but it's all `*Transformer` and we have BigDecimal conversions in both directions (indeed we do): we may end up with identical class names for different transformations (which can be resolved by using different package names).

> If we go with this scheme then perhaps we should rename TypeNormaliserFactory to something like jdbcToDbFitTransformerFactory (or jdbcToDbFitNormaliserFactory).

Maybe... But now I think we have more or less the same factory behavior in both cases (the factory doesn't make any decisions about what transformers are registered in it - it's up to the factory client). With some refactoring we may even get rid of `TypeNormaliserFactory` and keep a single class only.

> I do like the Normaliser part of the name.

Yes, I'm OK with that too. We "normalize" things into DbFit's format (this comes from JDBC but we also have Fit/FitNesse conversions from String to DbFit normalized types too).